### DISCUSSION TITLE
Add ability to use Loop Elements in custom widgets

### DISCUSSION THREAD

**Discussion by vladninja** [elementor/elementor#21257] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

In some cases, I need to create a custom widget with some controls, but don't want to create a some "fix" HTML for the render of this widget. The better way it's create a custom widget (like Loop Grid) with some controls and add possibility to the user to select loop element from the list and render the widget based on the loop element and controls.

### Describe the solution you'd like

Add control which 3-rd party developers may use for selecting a loop element from the list and apply the result on the Elementor editor and a website.

Something like:
```
$element->add_control(
    'loop_element',
    [
        'label' => __('Loop Element', 'elementor'),
        'label_block' => true,
        'type' => \Elementor\Controls_Manager::LOOP_ELEMENT_SELECT,
    ]
);
```

**Comment by bstrokim** [comment]
By "Loop Element," do you mean the Loop Item? If so, this is how I'm able to add a control for selecting a loop item:

```
// add loop item control
$options = array();

$posts = get_posts( array(
	'post_type'  => 'elementor_library',
	'posts_per_page'    => -1,
	'meta_query'	=> array(
		array(
			'key'     => '_elementor_source',
			'value'   => 'post',
			'compare' => '='
		),
		array(
			'key'     => '_elementor_template_type',
			'value'   => 'loop-item',
			'compare' => '='
		),
	)
) );

foreach ( $posts as $key => $post_ ) {
	$options[$post_->ID] = get_the_title($post_->ID);
}

$this->add_control(
	'loop_item_id',
	[
		'label' => esc_html__( 'Loop Item', 'my-plugin' ),
		'type' => \Elementor\Controls_Manager::SELECT2,
		'label_block' => true,
		'options' => $options
	]
);
```
Then I get the ID with:

```
$settings = $this->get_settings_for_display();
$my_loop_item = $settings['loop_item_id'];
```

My problem with this now is how to render it in my custom widget's loop.
